dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIFunctionToolCall.cs

```diff
@@ -84,4 +86,95 @@ public override string ToString()
 
         return sb.ToString();
     }
+
+    /// <summary>
+    /// Tracks tooling updates from streaming responses.
+    /// </summary>
+    /// <param name="update">The tool call update to incorporate.</param>
+    /// <param name="toolCallIdsByIndex">Lazily-initialized dictionary mapping indices to IDs.</param>
+    /// <param name="functionNamesByIndex">Lazily-initialized dictionary mapping indices to names.</param>
+    /// <param name="functionArgumentBuildersByIndex">Lazily-initialized dictionary mapping indices to arguments.</param>
+    internal static void TrackStreamingToolingUpdate(
+        StreamingToolCallUpdate? update,
+        ref Dictionary<int, string>? toolCallIdsByIndex,
+        ref Dictionary<int, string>? functionNamesByIndex,
+        ref Dictionary<int, StringBuilder>? functionArgumentBuildersByIndex)
```

Comment:
nit: Would it be better to have one Dictionary and a class to store the tool call id, function name and function arguments?

Reply:
If we expect only a small number of tool requests per interaction, your suggestion is better. If we expect a larger number, this is better.  It's probably the case that fewer is more likely.  If it's ok with you, though, let's hold off on changing it until we have more runway. Here I just copied the recommended approach from the Azure SDK readme.